### DISCUSSION TITLE
How do I create a pagination in a layout file?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#854] [Ask a question] [answered]
Hello!
I hope you had a nice Easter! 

Unfortunately I have a question again.
This time it is about the pagination. I do not understand what I am doing wrong. Maybe one of you has an idea.

I have installed a blog. This blog is displayed normally on the `blog-start.html`. Another page shows only the news. 
The file structure looks like this:

```
|----- content               
|    |----- 1-home
|    |----- 2-blog
|         |----- file-one.md
|         |----- file-two.md
|         | ...
|         |----- page.md    
|    |----- 3-news           
|         |----- file-one.md
|         |----- file-two.md
|         | ...
|         |----- page.md           
|----- media                    
|----- system
| ...

```
The header of the `page.md` in the `3-news` folder looks like this:

```
---
Title: News
Layout: news
LayoutNew: news
---
```
The header of the `page-new-news.md` that I put in the `shared` folder looks like this:
```
---
Title: News Name
Published: @datetime
Author: @username
Layout: newspost
Cat: news
---
```
The posts in my `news.html` are displayed with the following code:
(two posts are displayed for test purposes)

```
---- code before ----
<?php  foreach ($this->yellow->content->index()->filter("cat", "news")->sort("published", false)->paginate(2) as $page): ?>
---- more code ----
<?php endforeach ?>
---- more code ----
<?php $this->yellow->layout("pagination", $this->yellow->content->index()->filter("cat", "news")) ?>
---- code after ----
```
My problem is, that the pagination does not work.

`<?php $this->yellow->layout("pagination", $this->yellow->content->index()->filter("cat", "news")) ?>`

I hope I have not forgotten anything important. &#128556;
It would be great if someone has an idea.

**Comment by annaesvensson** [comment, marked as answer]
You are 50% there, just call `layout("pagination")` with paginated pages. When you call it with pages that have not been paginated, it doesn't know what to do. [Make a page collection](https://datenstrom.se/yellow/help/api-for-developers#yellow-page-collection), store it in a variable, use the variable in the foreach loop and the pagination.

Layout file for showing latest pages with pagination:

```
<?php $this->yellow->layout("header") ?>
<div class="content">
<div class="main" role="main">
<h1><?php echo $this->yellow->page->getHtml("titleContent") ?></h1>
<?php $pages = $this->yellow->content->index()->sort("modified", false)->paginate(10) ?>
<?php $this->yellow->page->setLastModified($pages->getModified()) ?>
<ul>
<?php foreach ($pages as $page): ?>
<li><?php echo $page->getHtml("title") ?></li>
<?php endforeach ?>
</ul>
<?php $this->yellow->layout("pagination", $pages) ?>
</div>
</div>
<?php $this->yellow->layout("footer") ?>
```

**Reply by PetersOtto** [reply]
Hello Anna!
Thank you very much for your help. This is how it works. In my case the solution looks like this:

```
---- code before ----
<?php $pages = $this->yellow->content->index()->filter("cat", "news")->sort("published", false)->paginate(2) ?>
<?php  foreach ($pages as $page): ?>
---- more code ----
<?php endforeach ?>
---- more code ----
<?php $this->yellow->layout("pagination", $pages) ?>
---- code after ----
```

I am very happy &#128522;
